bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/ThingUID.java

```diff
@@ -135,29 +135,11 @@ public ThingUID(String... segments) {
         super(segments);
     }
 
-    /**
-     * Returns the thing type id.
-     *
-     * @return thing type id
-     * @deprecated use {@link Thing#getThingTypeUID()} instead.
-     */
-    @Deprecated
-    public String getThingTypeId() {
-        String thingType = getSegment(1);
-        if (NO_THING_TYPE.equals(thingType)) {
-            return null;
-        } else {
-            return thingType;
-        }
-    }
-
     /**
      * Returns the thing type uid.
      *
      * @return thing type uid
-     * @deprecated use {@link Thing#getThingTypeUID()} instead.
```

Comment:
It's probably deprecated because this method always creates a new `ThingTypeUID`, whereas `Thing.getThingTypeUID()` returns an existing object. In a binding you typically already have a `Thing` object, so it will save CPU cycles and garbage (+ collection) by calling `Thing.getThingTypeUID()` and reusing the existing objects where possible.

Reply:
true ... and it can return `null` which` breaks other stuff like discovery ... (see e.g. https://github.com/openhab/openhab-core/pull/1414/commits/2d88d587e05d793b755aee4abfa349c8ce7803df).

Reply:
I finally found a way to remove it.

Reply:
Some more things to solve for the add-ons build: https://github.com/openhab/openhab-addons/issues/7402 and https://github.com/openhab/openhab-addons/pull/7403.